apps/builder/app/builder/features/workspace/workspace.tsx

```diff
@@ -45,6 +39,37 @@ const useSetWorkspaceRect = () => {
   return ref;
 };
 
+const getCanvasStyle = (
+  scale: number,
+  workspaceRect?: DOMRect,
+  canvasWidth?: number
+) => {
+  let canvasHeight;
+  let canvasLeft;
+
+  if (workspaceRect?.height) {
+    canvasHeight = workspaceRect.height / (scale / 100);
+  }
+
+  if (workspaceRect?.width && canvasWidth) {
+    canvasLeft = Math.max((workspaceRect.width - canvasWidth) / 2, 0);
```

Comment:
I have a feeling that canvasLeft>0 only if transform scale === 1, 
that allows to not calculate left position but use for canvas positioning ordinary Flex box with justify-content: center. 
Less calculus is always better